### ISSUE TITLE
10.7.5 - Direct play issues

### ISSUE THREAD

**Issue by Liquoriced** [jellyfin/jellyfin#5978]
After upgrading to 10.7.5, I'm unable to play anything without enabling the "Allow video playback that requires conversion without re-encoding" option. Before the upgrade, such option was not needed to play videos.

**System**
OS: Linux
Virtualization: Docker
Clients: Browser
Browser: Firefox 86
Jellyfin Version: 10.7.5
Playback: Direct Play
Installed Plugins: none
Reverse Proxy:none
Storage: local NAS

**To Reproduce**
1. Play any movie
2. Nothing happens
3. Enable "Allow video playback that requires conversion without re-encoding"
4. Movies play correctly

**Expected behavior**
I was able to play directly the same movies on the same devices (Kodi and MPV clients) before upgrading to 10.7.5.
Now all my movies only play if I enable the "Allow video playback that requires conversion without re-encoding" option.

**Logs**
[2021-05-05 15:41:31.072 +08:00] [INF] [42] Jellyfin.Api.Helpers.MediaInfoHelper: User policy for "User". EnablePlaybackRemuxing: False EnableVideoPlaybackTranscoding: False EnableAudioPlaybackTranscoding: False
[2021-05-05 15:41:31.072 +08:00] [INF] [42] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Kodi", Path: "\\192.168.1.101\video\movies\10.Cloverfield.Lane.2016.COMPLETE.UHD.BLURAY-COASTER{{3aBCOE0mAtdeTv2PF1W3FmE2E}}.mkv", isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[2021-05-05 15:41:40.203 +08:00] [WRN] [34] Emby.Server.Implementations.HttpServer.WebSocketConnection: WS "192.168.1.104" error receiving data: "The remote party closed the WebSocket connection without completing the close handshake."
[2021-05-05 15:41:40.401 +08:00] [INF] [34] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "192.168.1.104" closed

**Comment by cvium** [comment]
Please enable transcoding, start playback and upload the Playback Data/Info

**Comment by Liquoriced** [comment]
Is this what you need?

[2021-05-05 16:20:40.982 +08:00] [WRN] [67] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from "http://192.168.1.101:8096/Sessions/Playing/Progress" to "192.168.1.104" in 0:00:00.535959 with Status Code 204
[2021-05-05 16:20:55.677 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: session.RemoteEndPoint : "192.168.1.104"
[2021-05-05 16:20:55.678 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : event_playing_id     = "9378834644d8646c457ea9679c6d4efd"
[2021-05-05 16:20:55.678 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : event_user_id        = "f0d28b45a3284512a2aecaadae100aba"
[2021-05-05 16:20:55.678 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : event_user_id_int    = 1
[2021-05-05 16:20:55.679 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : session_playing_id   = "9378834644d8646c457ea9679c6d4efd"
[2021-05-05 16:20:55.679 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : session_user_id      = "f0d28b45a3284512a2aecaadae100aba"
[2021-05-05 16:20:55.679 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : play_method          = "DirectStream"
[2021-05-05 16:20:55.679 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : e.ClientName         = "Kodi"
[2021-05-05 16:20:55.680 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : e.DeviceName         = "Kodi (192.168.1.104)"
[2021-05-05 16:20:55.680 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : ItemName             = "10 Cloverfield Lane"
[2021-05-05 16:20:55.680 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : ItemId               = "9378834644d8646c457ea9679c6d4efd"
[2021-05-05 16:20:55.680 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : ItemType             = "Movie"
[2021-05-05 16:20:55.680 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : All matches, playback registered
[2021-05-05 16:20:55.681 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: Playback tracker found, adding playback info : "b73f5570-58bc-4014-8c37-f0009df51643-f0d28b45a3284512a2aecaadae100aba-9378834644d8646c457ea9679c6d4efd"
[2021-05-05 16:20:55.681 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: Saving playback tracking activity in DB
[2021-05-05 16:20:55.911 +08:00] [INF] [4] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : Exited

**Comment by cvium** [comment]
No, I need something like this
![image](https://user-images.githubusercontent.com/675118/117123286-2194d100-ad97-11eb-892a-c7233f3e90a2.png)

**Comment by Liquoriced** [comment]
Got it - here it is, below.

One correction from my earlier post: rather than proper transcoding, this is the option that I need to enable to be able to play videos under 10.7.5: "Allow video playback that requires conversion without re-encoding".
I'll edit my initial message to avoid confusion.

![Screenshot_20210505-184552_Jellyfin](https://user-images.githubusercontent.com/80312124/117131033-77876a00-add3-11eb-83b3-764aa61bc1f3.jpg)

**Comment by Peobran** [comment]
Hello,
I can't play video in direct play anymore too.
The problem is that I play everything in direct play via Kodi or the browser.
I have a very small processor.
I can't use my server anymore

**Comment by Shadowghost** [comment]
Enable `Allow video playback that requires conversion without re-encoding`. The setting is disabled by default but wasn't enforced until 10.7.5 - it is now, so you most likely just need to enable it to make `DirectStream` work again.

Most of you are most likely using `DirectStream` and not `DirectPlay` anyway. The latter plays the source file 1:1, the former repackages it into something the client can handle.

**Comment by SteveDinn** [comment]
I had the same problem with 10.7.5. nothing would play unless I enabled
that option for the user. Things still direct played though, so it seems
like that option is being checked even though the file could be direct
played (i.e., it didn't require any conversion).

I had to roll back to 10.7.2 in the short term.

**Comment by MovieWiz** [comment]
Like SteveDinn, I could get nothing to play unless I enabled "Allow video playback that requires transcoding".  I have windows server and 1.6 Jellyfin Media Player.  Once I enabled the option, everything direct played again like it did in 10.7.2.  But I don't like having it enabled because negotiation between the server and the AndroidTV app on my 2020 Fire Stick Lite often doesn't work optimally and video will get transcoded that would direct play just fine.  So I've gone back to 10.7.2 so that I can leave "Allow video playback that requires transcoding" disabled and have videos continue to direct play on my FireStick.

There is a separate option for transcoding audio.  So I did not understand Shadowghost's comment that the "Allow video playback that requires transcoding" option is being enforced.  Disabling that option should not stop Direct Play from working.  It should not even stop Direct Streaming from working since direct streaming is only transcoding the audio. and I have audio transcoding enabled.

Has the meaning/purpose of ""Allow video playback that requires transcoding"" changed in 10.7.5?

**Comment by Peobran** [comment]
Hello,
After enabling `Allow video playback that requires conversion without re-encoding` It's working.
But it was disabled before in 17.2 for me and it worked before.
I want to Direct Play, not Direct Stream. This is why I use Kodi. And I can't do it anymore.

**Comment by Shadowghost** [comment]
I agree that Kodi should `DirectPlay` and not `DirectStream`. But as I said before: `It worked before` is not a valide argument since we didn't enforce the setting in any kind before 10.7.5 - it was always enabled.

Afaik the logs should show why something is DirectStream instead of DirectPlay.

**Comment by Peobran** [comment]
So what you're saying is that even with `Allow video playback that requires conversion without re-encoding` disabled before 10.7.5, it used to `DirectStream` instead of running `DirectPlay` ?

**Comment by MovieWiz** [comment]
I'm still really confused @Shadowghost.  It seems that you're saying that the "Allow video playback that requires transcoding" option is now required or enforced even when something will Direct Play (or Direct Stream which only re-encodes the audio).  I have Jellyfin Media Play and Jellyfin server on my Windows box and everything Direct Plays.  I can see that in the Dashboard.  Plus I know it's direct playing since the video file is an MP4 with H264 and AC3 5.1.

So I don't understand why having ""Allow video playback that requires transcoding" enabled in 10.7.5 is required. If so, I don't understand what the meaning/purpose of this option is anymore.

When you said Direct Stream causing the video/audio to be repackaged, did you mean "Remux"?  There is an option for allowing/disallowing that as well.

I'm concerned because it seemed like you were saying that there is no bug here when it seems to me that Jellyfin Media Player should be able to Direct Play just fine without this option enabled.

**Comment by SteveDinn** [comment]
@MovieWiz I believe we're talking about the option `Allow video playback that requires conversion without re-encoding`, **NOT** `Allow video playback that requires transcoding`.

In point form:
* All my video is direct-playable on my devices.
* In 10.7.2, in that range of checkboxes, the only one I had checked is `Allow media playback`.
* After upgrading to 10.7.5, nothing played with only that option selected.
* Reading a bit, I tried enabling the `Allow video playback that requires conversion without re-encoding` option.
* Videos play now, but they still direct-PLAY (not direct-STREAM). So I don't understand why the option was necessary in the first place, and I think this is the bug.

TL;DR As weird as it seems, I believe the bug here is not actually with playback, but simply requiring the option to be selected when it doesn't really need to be.

**Comment by Peobran** [comment]
@SteveDinn How do you know it DirectPlay and not DirectStream ?

**Comment by SteveDinn** [comment]
> @SteveDinn How do you know it DirectPlay and not DirectStream ?

@Peobran I'm looking in the Dashboard view and it tells you for all playing media, how it's playing it, Direct Playing, Direct Streaming, or Transcoding. Even with that option selected (which would lead me to believe that it's going to try to Direct Stream, it is still telling me that it's Direct Playing.

**Comment by MovieWiz** [comment]
Really sorry @SteveDinn for confusing the matter.  Don't know how I read that wrong.  I deleted my comments from the thread.

Yes, it appears those options need to be enabled when they shouldn't need to be.  If the item is going to be direct played, it doesn't make sense that either option needs to be enabled.

**Comment by SteveDinn** [comment]
In some ways, there is an improvement here though. I had those options disabled because, like you, I wanted to force direct playing when (in 10.7.2) it was incorrectly doing audio transcoding or remuxing when it wasn't necessary. It now (in 10.7.5) seems to be choosing direct playing even when that option is selected :)

**Comment by da-anda** [comment]
not sure if it's the same issue or not, but I have the following enabled in my user profile

- [x] "Allow media playback"
- [x] "Allow audio playback that requires transcoding"
- [ ] "Allow video playback that requires transcoding"
- [x] "Allow video playback that requires conversion without re-encoding"
- [ ] "Force transcoding of remote media sources such as Live TV"

And I am not able to watch anything in Firefox via the webclient that requires remuxing (regular h264 SDR video). Only when I enable video transcoding, then direct stream via remuxing and video stream copy seems to be working.

**Comment by MallemIsmail** [comment]
Same exacte profile config of @da-anda above me and the same problem, i used to play everything now unless i enable video transcoding.

**Comment by silic0ns0ul** [comment]
It is weird that files that did not require video transcoding before, now require this setting enabled to play. But what's worse is that it actually starts video transcoding and my underpowered CPU is definitely not up to the task. (typically it's the audio that needs transcoding and not the video.) So basically, unless I roll back, I am unable to use my jellyfin server now.

**Comment by Elaws** [comment]
> ✓ "Allow media playback"
> ✓ "Allow audio playback that requires transcoding"
> ✓ "Allow video playback that requires transcoding"
> ✓ "Allow video playback that requires conversion without re-encoding"
> × "Force transcoding of remote media sources such as Live TV"

Impossible to play some videos accessed from Android client over internet access (set up with Caddy), reverting to 10.7.2 didn't help.

Message in log file :
 ```
Profile: "Unknown Profile", Path: "PATH_TO_VIDEO", isEligibleForDirectPlay: True, isEligibleForDirectStream: False
Bitrate exceeds DirectStream limit: media bitrate: 1383876, max bitrate: 459242
```

**Comment by Elaws** [comment]
Hi,

Are there any news on this issue ? Could something similar to [jellyfin/jellyfin-kodi#506 ](https://github.com/jellyfin/jellyfin-kodi/pull/506) be implemented in jellyfin server as a temporary workaround to remove this bitrate limit ?

Thanks

**Comment by AFulgens** [comment]
Agree with @Elaws. For me the proper (?) long-term solution seems to be to make the `max bitrate` limit of DirectPlay configurable in the UI with the current values set as default.

**Comment by TwistTheNeil** [comment]
Having the same issue with 10.7.7

```
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Controllers.MediaInfoController: GetPostedPlaybackInfo profile: {"Name": "Jellyfin MPV Shim", "Id": null, "Identification": null, "FriendlyName": null, "Manufacturer": null, "ManufacturerUrl": null, "ModelName": null, "ModelDescription": null, "ModelNumber": null, "ModelUrl": null, "SerialNumber": null, "EnableAlbumArtInDidl": false, "EnableSingleAlbumArtLimit": false, "EnableSingleSubtitleLimit": false, "SupportedMediaTypes": "Audio,Photo,Video", "UserId": null, "AlbumArtPn": null, "MaxAlbumArtWidth": 0, "MaxAlbumArtHeight": 0, "MaxIconWidth": null, "MaxIconHeight": null, "MaxStreamingBitrate": 2147483000, "MaxStaticBitrate": 8000000, "MusicStreamingTranscodingBitrate": 1280000, "MaxStaticMusicBitrate": null, "SonyAggregationFlags": null, "ProtocolInfo": null, "TimelineOffsetSeconds": 5, "RequiresPlainVideoItems": false, "RequiresPlainFolders": false, "EnableMSMediaReceiverRegistrar": false, "IgnoreTranscodeByteRangeRequests": false, "XmlRootAttributes": [], "DirectPlayProfiles": [{"Container": null, "AudioCodec": null, "VideoCodec": null, "Type": "Video", "$type": "DirectPlayProfile"}, {"Container": null, "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": null, "AudioCodec": null, "VideoCodec": null, "Type": "Photo", "$type": "DirectPlayProfile"}], "TranscodingProfiles": [{"Container": null, "Type": "Audio", "VideoCodec": null, "AudioCodec": null, "Protocol": null, "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": null, "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "ts", "Type": "Video", "VideoCodec": "h264,h265,hevc,mpeg4,mpeg2video", "AudioCodec": "aac,mp3,ac3,opus,flac,vorbis", "Protocol": "hls", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "6", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "jpeg", "Type": "Photo", "VideoCodec": null, "AudioCodec": null, "Protocol": null, "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": null, "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}], "ContainerProfiles": [], "CodecProfiles": [], "ResponseProfiles": [], "SubtitleProfiles": [{"Format": "srt", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "srt", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "ass", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "ass", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "sub", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "sub", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "ssa", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "ssa", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "smi", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "smi", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "pgssub", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "dvdsub", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "pgs", "Method": "Embed", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}], "$type": "DeviceProfile"}
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: User policy for MyJellyfinUser. EnablePlaybackRemuxing: False EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: Bitrate exceeds DirectPlay limit: media bitrate: 69238197, max bitrate: 8000000
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: Jellyfin MPV Shim, Path: /media/video.mkv, isEligibleForDirectPlay: False, isEligibleForDirectStream: True
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: Bitrate exceeds DirectPlay limit: media bitrate: 69238197, max bitrate: 8000000
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: Jellyfin MPV Shim, Path: /media/video.mkv, isEligibleForDirectPlay: False, isEligibleForDirectStream: True
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: Bitrate exceeds DirectPlay limit: media bitrate: 69238197, max bitrate: 8000000
jellyfin    | [04:45:21] [INF] [50] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: Jellyfin MPV Shim, Path: /media/video.mkv, isEligibleForDirectPlay: False, isEligibleForDirectStream: True

```

**Comment by sears-s** [comment]
For anyone still waiting to update, I recently updated to the latest version and DirectPlay seems to be working again as before. I previously was sticking to 10.7.2 to fix the issue, but my client started telling me the server version was too old. Not sure which exact version fixed the issue.
